packages/liveblocks-react/src/factory.tsx

```diff
@@ -433,54 +548,159 @@ export function createRoomContext<
 
   function useMyPresence(): [
     TPresence,
-    (overrides: Partial<TPresence>, options?: { addToHistory: boolean }) => void
+    (patch: Partial<TPresence>, options?: { addToHistory: boolean }) => void
   ] {
     const room = useRoom();
     const presence = room.getPresence();
     const rerender = useRerender();
+    const setPresence = room.updatePresence;
 
-    React.useEffect(() => {
-      const unsubscribe = room.subscribe("my-presence", rerender);
-      return () => {
-        unsubscribe();
-      };
-    }, [room, rerender]);
-
-    const setPresence = React.useCallback(
-      (overrides: Partial<TPresence>, options?: { addToHistory: boolean }) =>
-        room.updatePresence(overrides, options),
-      [room]
-    );
+    React.useEffect(() => room.events.me.subscribe(rerender), [room, rerender]);
 
     return [presence, setPresence];
   }
 
   function useUpdateMyPresence(): (
-    overrides: Partial<TPresence>,
+    patch: Partial<TPresence>,
     options?: { addToHistory: boolean }
   ) => void {
-    const room = useRoom();
+    return useRoom().updatePresence;
+  }
 
-    return React.useCallback(
-      (overrides: Partial<TPresence>, options?: { addToHistory: boolean }) => {
-        room.updatePresence(overrides, options);
-      },
-      [room]
+  function useOthers(): Others<TPresence, TUserMeta>;
+  function useOthers<T>(
+    selector: (others: Others<TPresence, TUserMeta>) => T,
+    isEqual?: (prev: T, curr: T) => boolean
+  ): T;
+  function useOthers<T>(
+    selector?: (others: Others<TPresence, TUserMeta>) => T,
+    isEqual?: (prev: T, curr: T) => boolean
+  ): T | Others<TPresence, TUserMeta> {
+    const room = useRoom();
+    const subscribe = room.events.others.subscribe;
+    const getSnapshot = room.getOthers;
+    const getServerSnapshot = getEmptyOthers;
+    return useSyncExternalStoreWithSelector(
+      subscribe,
+      getSnapshot,
+      getServerSnapshot,
+      selector ?? (identity as (others: Others<TPresence, TUserMeta>) => T),
+      isEqual
     );
   }
 
-  function useOthers(): Others<TPresence, TUserMeta> {
-    const room = useRoom();
-    const rerender = useRerender();
+  function useOtherIds(): readonly number[]; // TODO: Change to ConnectionID for clarity?
+  function useOtherIds<T>(
```

Comment:
`useOtherIds` relies on `User.connectionId` which is confusing because we have an 'id' property on `User`. Not sure if simplifying the name is worth the potential confusion. 🤔

Reply:
I love this hook, but still hate the name. Also dislike `useOthersConnectionIds`. Looking for a name that stresses the iterating/mapping/subselection nature—while pairing it with the connection IDs.

Any of these coming closer to a good name?
- useOtherForEach
- useEachOther (funny English ruins this one, unfortunately)
- useOthersMapped
- useMapOverOthers
- useSelectOthers
- useOthersData
- usePickFromOthers

Reply:
Naming proposal:

- We ditch the `useOtherIds()` hook that's now in 0.18.0-beta2.
- We split its two overloads into two new hooks:
  - `useConnectionIds(): number[]`
  - `useOthersMapped(itemSelector, isEqual): { connectionId: number; data: T }[]`

Reply:
I like the split in two hooks. Personally, I'm not a fan of the name `userOthersMapped`. But I don't have good ideas for a better name. The best name for me is `useOthersData`.

To give more ideas:
- useOthersAsIterable
- useSelectorOthers

Reply:
usePluckOthers

Reply:
How about `useOthers2` and in the doc string issue a naming contest? 😅

Reply:
You have my vote for `useOthersData` !

Reply:
Not easy to find the right name for this hook 😬 

My 2 cents on the proposed solutions:
- Not showing the word "connectionId" in the name forces developers to look at the return type or the documentation to understand what the hook is doing. I think this is not great for discoverability reasons, especially because we re-export them. I understand that putting such a long word in a name might not be appealing, but I think explicitness is better in this case.
- IMO, `selector` and `mapped` words are not explicit enough because `useOthers` is also taking a selector.

Taking these points into consideration, I would go with `useOthersWithConnectionId`. A tad too long but I think it's worth it to avoid the confusion with `useOthers(selector)`.

I also considered using a special selector like this:

```ts
useOthers(withConnectionId(user => user.presence.cursor))
```

But even if more composable, I think it's too indirect.

Reply:
> _Taking these points into consideration, I would go with `useOthersWithConnectionId`._

I can see that argument. My gripe with that name is that (1) `useOthers` also returns a value that has connection IDs, so initial reactions may be "why the `WithConnectionId` part" and (2) to me, although it _also_ returns connection IDs, the main purpose is the derived `data` you want to select. That's why it exists mainly. The connection ID is only there to support typical React usage of it, but it isn't the main thing. That's why I've been leaning towards `useOthersWithData` more. Especially now that we're splitting off the `useConnectionIds()` as a separate (and clearly named!) hook.

So, the ultimate compromise here between your both proposals might actually be the name I prefer most…

`useOthersData` + `useOthersWithConnectionId` = `useOthersWithData` ❤️ ?

Reply:
What about `useConnectionIds` and `useConnectionData`? The connection feels a little more obvious there, and I prefer this. My second choice is `useConnectionIds` and `useOthersWithConnectionId`, even if it's long, because it's quite clear what it is.

---

Alternatively, what if we had a hook that returned selectors, instead of ids:
```
function Parent() {
  // List of selector functions for each other user
  const selectors = useOthersSelectors();

  // Functions are stable and can be used as map keys (is this allowed in React?)
  return (
    {selectors.map((useOther) => 
      <Child key={useOther} useOther={useOther} />
    )}
  );
}


function Child({ useOther }) {
  const cursor = useOther((other) => other.presence.cursor));
}
```

Could also return ids and selectors:
```
function Parent() {
  // List of { id, selector } objects for each other user
  const selectors = useOthersSelectors();

  // Use id as map key
  return (
    {selectors.map((other) => 
      <Child key={other.id} useOther={other.selector} />
    )}
  );
}


function Child({ useOther }) {
  const cursor = useOther((other) => other.presence.cursor));
}
```

Reply:
@CTNicholas 💡! That's really clever! I really like that we don't have to transfer any "connectionId" from the parent to the children components. I have never seen a single library where you pass a hook as a property so I'm curious to know how you got that idea. Is it coming from you or did you see this pattern somewhere else?

Reply:
@GuillaumeSalles Thanks—I just enjoy coming up with unconventional API ideas, I've never seen it before! Definitely wondering if there's a reason we haven't seen it.